corehq/preindex/__init__.py

```diff
@@ -39,6 +40,12 @@ def get_designs(self):
         raise NotImplementedError()
 
     def sync_design_docs(self, temp=None):
+        if self.synced:
+            # workaround emit_post_sync_signal called twice
+            # https://code.djangoproject.com/ticket/17977
+            # this is to speed up test initialization
```

Comment:
Can you not use this approach: https://docs.djangoproject.com/en/1.8/topics/signals/#preventing-duplicate-signals